superset/views/core.py

```diff
@@ -1785,6 +1786,11 @@ def publish(  # pylint: disable=no-self-use
     @has_access
     @expose("/dashboard/<dashboard_id_or_slug>/")
     @event_logger.log_this_with_extra_payload
+    @check_dashboard_access(
+        on_error=lambda self, ex: Response(
+            utils.error_msg_from_exception(ex), status=403
+        )
+    )
```

Comment:
Maybe you can do away with the decorator for now to avoid calling `Dashboard.get` twice, or the decorator could actually update functional signature to pass the already found `dash` object to the wrapped function.

Reply:
@ktmud look again 
I've passed the dashboard queried from the decorator back to the original func.
we could have a dedicated decorator just for querying the dashboard but I think that would be an overhead right now